core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/Flow.java

```diff
@@ -4,4 +4,7 @@
     void initialize();
     void sendEvent(String key, Object object);
     FlowState getCurrentState();
+    String getFlowId();
+    void setFlowFailed(boolean failed);
```

Comment:
Parameter not necessary, because we call with a true value if there is some error, and i couldn't imagine scenario where a failed flow become non failed later.